packages/node_modules/@ciscospark/widget-space/README.md

```diff
@@ -131,6 +131,16 @@ When loading the widgets there are some configuration options you can provide:
 | `toPersonEmail` | `data-to-person-email` | Email of the message recipient |
 | `toPersonId` | `data-to-person-id` | User Id of the message recipient |
 
+### External Control
+
+The space widget allows for developers to control actions and behaviors via properties. These properties, when set, will make the widget perform certain actions, like switching to different activities.
+
+**Note:** these actions are triggered by the properties changing from empty to populated. This means to perform a second action, the property will need to be set back to empty before changing.
```

Comment:
Since we don't have two way data binding of props in react, there wasn't a smooth way I could find to properly control the widget otherwise. For example, if you set the prop to tell it to switch to the "meet" widget, and then the user manually switches back to the "message" widget, your `setCurrentActivity` prop would still be set externally to "meet". We won't get a `componentWillReceiveProps` event if you try to set it to "meet" again until you empty the prop, then change it back to "meet". 

I'm definitely up for a different method, but I haven't heard of any from the other developers I've asked this about.

Reply:
I can't think of a better solution, so I think we may have to stick with this until if/when we find something better.